src/components/SignupForm.js

```diff
@@ -0,0 +1,240 @@
+                setTimerOn(false);
+            }
+        })
+        .catch((err) => {
+            // // 인증번호 불일치
+            // alert('인증번호가 만료되었습니다. 재발급해주세요.');
+            // setTimeover(true);
+            // setTimerOn(false);
+        });
+    }
+
+    // 회원가입 입력값 예외 처리
+    const checkSignupData = () => {
+
+        if(Object.values(signupInfo).includes('')) {
+            alert('입력되지 않은 값이 있습니다.');
+            return;
+        }
+
+        if(!isValidPassword(signupInfo.password)) {
+            alert('비밀번호를 8자리 이상 입력해 주세요.')
+            return;
+        }
+
+        if(signupInfo.password !== checkPassword) {
+            alert('비밀번호가 일치하지 않습니다. 다시 입력해 주세요.');
+            return;
+        }
+
+        if(!isValidCarNumber(carNumber)) {
+            alert('차량번호를 올바르게 입력해 주세요.')
+            return;
+        }
+
+        if(!completeCertification) {
+            alert('인증이 완료되지 않았습니다.');
+            return;
+        }
+    }
+
+    // 회원가입 통신
+    const onClickSignup = () => {
+
+        checkSignupData();
+        
+        let url = 'company';
+        if(signupInfo.auth === 'USER') {
+            url = 'new';
+            signupInfo.car = carNumber;
+        }
+
+        axios.post(`${process.env.REACT_APP_PROXY}/member/${url}`, signupInfo)
+        .then((res) => {
+            navigate("/login");
+        })
+        .catch((err) => {
+            console.log(err);
+        });
+
+    }
+
+    return (
+        <div className='Form' id='Signup'>
+            <div id='typeRadio'>
+                <input 
+                    type='radio' 
+                    value='USER' 
+                    name='auth' 
+                    checked={signupInfo.auth === 'USER'} 
+                    onChange={handleSignupData}/>
+                    &nbsp;&nbsp;<label>사용자</label> 
+                    &nbsp;&nbsp;&nbsp;
+                <input 
+                    type='radio' 
+                    value='MANAGER' 
+                    name='auth' 
+                    checked={signupInfo.auth === 'MANAGER'} 
+                    onChange={handleSignupData}/>
+                    &nbsp;&nbsp;<label>관리자</label>
+            </div>
+            <input 
+                    type='text' 
+                    value={signupInfo.name} 
+                    name='name' 
+                    onChange={handleSignupData}
+                    placeholder='  이름을 입력하세요'/>
+            <div id='signupBox'>
+                <input 
+                    value={signupInfo.email} 
+                    type='text' 
+                    name='email' 
+                    onChange={handleSignupData} 
+                    placeholder='  e-mail을 입력하세요'/>
+                {
+                    timerOn ? <Timer/> : (
+                    timeover ? 
+                    <button id='certifBtn' onClick={handleReCertification}>재발급받기</button> :
+                    <button id='certifBtn' onClick={handleCertification}>인증받기</button>)
+                }
```

Comment:
Repeated use of ternary operator reduces readability.

It is recommended to add an if statement by subtracting it as a function.